mitmproxy/proxy/protocol/http.py

```diff
@@ -327,7 +330,15 @@ def _process_flow(self, f):
                 )
 
                 def get_response():
```

Comment:
`get_response` can be called twice (see [here](https://github.com/mitmproxy/mitmproxy/pull/2369/files#diff-1a48d22b11223fa300cac1e75cbcfafcR351) for the reasoning). If `stream` is set to True, the second send will have no body, because we have already depleted the body generator on the first attempt.

Unfortunately this is really hard to fix. I see two possible ways forward:

1. Store all chunks _somewhere_ so that we can play them again. That'd be nice as we then could always store the content while streaming, but that also adds substantial complexity.
2. Figure out a better way to see if the socket is closed. Maybe it's enough to just try reading from it with MSG_PEEK. Alternatively, maybe we can make `get_response` a `send_headers` function, and then do everything else (sending body + receiving response) outside of the try-except-repeat block. I am not sure if any of this works, we need to test that.

I think the second option is the generally nicer way (we can still implement 1 later), so we should probably try and improve that in a separate PR.